examples/frigate/frigate.py

```diff
@@ -0,0 +1,143 @@
+    context: RuntimeContext = srv.mwcore["context"]
+    service_config = context.get_service_config("apprise-ntfy")
+    filename_template = service_config.get("filename_template")
+
+    # Decode JSON message.
+    after = json.loads(data['payload'])['after']
+
+    # Collect details from inbound Frigate event.
+    event = FrigateEvent(
+        time=datetime.fromtimestamp(after['frame_time']),
+        camera=after['camera'],
+        label=after['sub_label'] or after['label'],
+        current_zones=after['current_zones'],
+        entered_zones=after['entered_zones'],
+    )
+
+    # Interpolate event data into attachment filename template.
+    attach_filename = filename_template.format(**event.to_dict())
+
+    # Compute parameters for outbound Apprise / Ntfy URL.
+    ntfy_parameters = NtfyParameters(
+        title=f"{event.label} entered {event.entered_zones_str}",
+        format=f"In zones {event.current_zones_str} at {event.time}",
+        click=f"https://frigate/events?camera={event.camera}&label={event.label}&zone={event.entered_zones[0]}",
+        attach=attach_filename,
+    )
+    return ntfy_parameters.to_dict()
+
+
+def frigate_events_filter(topic, message, section, srv: Service):
+    """
+    mqttwarn filter function to only use Frigate events of type `new`.
+
+    Additionally, validate more details within the event message,
+    specifically the `after` section. For example, skip false positives.
+
+    :return: True if message should be filtered, i.e. notification should be skipped.
+    """
+    try:
+        message = json.loads(message)
+    except json.JSONDecodeError as e:
+        srv.logging.warning(f"Can't parse Frigate event message: {e}")
+        return True
+
+    # ignore ending messages
+    if message.get('type', None) == 'end':
+        return True
+
+    # payload must have 'after' key
+    elif "after" not in message:
+        srv.logging.warning("Frigate event skipped: 'after' missing from payload")
+        return True
+
+    after = message.get('after')
+
+    nonempty_fields = ['false_positive', 'camera', 'label', 'current_zones', 'entered_zones', 'frame_time']
+    for field in nonempty_fields:
+
+        # Validate field exists.
+        if field not in after:
+            srv.logging.warning(f"Frigate event skipped, missing field: {field}")
+            return True
+
+        value = after.get(field)
+
+        # We can ignore if `current_zones` is empty.
+        if field == "current_zones":
+            continue
+
+        # Check if it's a false positive.
+        if field == "false_positive":
+            if value is True:
+                srv.logging.warning("Frigate event skipped, it is a false positive")
+                return True
+            else:
+                continue
+
+        # All other keys should be present and have values.
+        if not value:
+            srv.logging.warning(f"Frigate event skipped, field is empty: {field}")
+            return True
+
```

Comment:
End users (such as myself) may also want to further filter `update` messages by checking if `current_zones` is equal to `entered_zones`; this can happen if the snapshot is updated or its state has changed, but the object has not moved to a new zone. For example, you might add something like this:
```python
if message['type'] == 'update' and after['current_zones'] == after['entered_zones']:
    srv.logging.warning("Frigate event skipped: Object didn't enter new zone")
```